### PR TITLE
[fix] Prevent false positives in rpmdeps inspection for modules

### PR DESCRIPTION
**Pull request by jchecahi**
Ensure that, when requesting an explicit package requires for a shared library, both provider and requester packages come from the same SRPM.

This PR aims to prevent these false possitives described in #1246.

### CODE REVIEW

**Comment by jchecahi** [comment]
Tested this with the same command described in issue #1246, the inspection no longer reports false possitives:
```
# ./build/src/rpminspect -c /usr/share/rpminspect/redhat.yaml -T rpmdeps --arch aarch64 llvm-toolset-rhel8-8090020230803191529.b46abd14
[...]
rpmdeps:                                                                                                                                                                                                                                                                        
--------                                                                                                                                                                                                                                                                        
Result: OK                                                                                                                                                                                                                                                                      
[...]
```

File: lib/inspect_rpmdeps.c [hunk 1]
```diff
@@ -260,6 +263,14 @@ static bool check_explicit_lib_deps(struct rpminspect *ri, Header h, deprule_lis
                 continue;
             }
 
+            /*
+             * issue 1246: Prevent false positives when the library is provided
+             * by a subpackage coming from another SRPM.
+             */
+            peer_srpm = headerGetString(peer->after_hdr, RPMTAG_SOURCERPM);
+            if (peer_srpm && strcmp(peer_srpm, srpm)) {
```
Comment (dcantrell) [inline lib/inspect_rpmdeps.c:271]: I think this is a suitable workaround now until I can get better cross-RPM dependency handling for modules.  I would prefer that we also guard this test to ri->buildtype as KOJI_BUILD_MODULE.

**Review by dcantrell — CHANGES_REQUESTED** [review]
I think this is a reasonable way to deal with dependencies that cross RPMs in modules for now, but I would like to restrict it to KOJI_BUILD_MODULE types.

File: lib/inspect_rpmdeps.c (hunk 1 above)
Comment (jchecahi) [reply lib/inspect_rpmdeps.c]: Thanks for checking. That guard makes sense, I've updated the PR with the guard.

File: lib/inspect_rpmdeps.c [hunk 2]
```diff
@@ -260,6 +263,15 @@ static bool check_explicit_lib_deps(struct rpminspect *ri, Header h, deprule_lis
                 continue;
             }
 
+            /*
+             * issue 1246: Prevent false positives when the library is provided
+             * by a subpackage coming from another SRPM.
+             */
+            peer_srpm = headerGetString(peer->after_hdr, RPMTAG_SOURCERPM);
```
Comment (dcantrell) [inline lib/inspect_rpmdeps.c:270]: Blank line after line 270, please.

File: lib/inspect_rpmdeps.c [hunk 3]
```diff
@@ -260,6 +263,15 @@ static bool check_explicit_lib_deps(struct rpminspect *ri, Header h, deprule_lis
...
+            if (ri->buildtype == KOJI_BUILD_MODULE
```
Comment (dcantrell) [inline lib/inspect_rpmdeps.c:271]: Join lines 271 and 272, please.

File: lib/inspect_rpmdeps.c [hunk 4]
```diff
@@ -260,6 +263,15 @@ static bool check_explicit_lib_deps(struct rpminspect *ri, Header h, deprule_lis
...
+                && peer_srpm && strcmp(peer_srpm, srpm)) {
+                continue;
+            }
```
Comment (dcantrell) [inline lib/inspect_rpmdeps.c:274]: Blank line after 274, please.

File: lib/inspect_rpmdeps.c (hunk 2 above)
Comment (jchecahi) [reply lib/inspect_rpmdeps.c]: I also moved the comment immediately above the condition check, it makes more sense.

**Review by dcantrell — APPROVED** [review]

**Pull request merged**